lib/events.js

```diff
@@ -356,8 +379,11 @@ EventEmitter.prototype.removeListener =
     function removeListener(type, listener) {
       var list, events, position, i, originalListener;
 
-      if (typeof listener !== 'function')
-        throw new TypeError('"listener" argument must be a function');
+      if (typeof listener !== 'function') {
+        const errors = lazyErrors();
```

Comment:
Do you know where the circular reference is coming from? If so - is there any possibility in untying that?
I personally also prefer using a simple if a lot over the function version. It is still a minor overhead of using the function while there is no faster check than a simple if to undefined.

Reply:
It's not a circular reference that is the concern. The `events` module is one of the very first modules loaded in core (because it is needed by `process`) and as such needs to be loaded before anything else is loaded.